scripts/publish.js

```diff
@@ -66,29 +81,18 @@ async function main() {
            * hasn't already been published
            */
           async (dirent) => {
-            if (!dirent.parent || !dirent.isDirectory()) {
-              return
-            }
-            const { name, version, private } = JSON.parse(
-              await fs.readFile(
-                path.join(
-                  ROOT,
-                  dirent.parent.name,
-                  dirent.name,
-                  'package.json'
-                ),
-                'utf8'
-              )
+            const { name, version, private } = await chillJSONParse(
+              fs.readFile(path.join(dirent.fullpath(), 'package.json'), 'utf8')
             )
-            if (private) {
+            if (private || !name) {
               return
             }
 
-            const { stdout } = await exec(`npm view ${name} versions --json`, {
-              cwd: ROOT,
-            })
-
-            const versions = JSON.parse(stdout)
+            const versions = await chillJSONParse(
```

Comment:
why mix await and then/catch?  these are all wrapped in a `Promise.all()`, so async operations should not happen in serial anyhow

Reply:
it was for readability. the function passed to map is async, so this will not turn anything into a sequence.